src/main/java/org/example/learningsystem/multitenancy/context/TenantContext.java

```diff
@@ -0,0 +1,42 @@
+package org.example.learningsystem.multitenancy.context;
+
+import lombok.NoArgsConstructor;
+
+import static lombok.AccessLevel.PRIVATE;
+
+/**
+ * Thread-local storage for current tenant information.
+ */
+@NoArgsConstructor(access = PRIVATE)
+public class TenantContext {
```

Comment:
It's better to name RequestContext, to make it more abstract. And here we will be able to put any data related to request (e.g. current tenant, ip address of caller and so on)